### PR TITLE
[ACM-6773]: Updating kubevirt docs

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-6773

This PR adds new documentation to the KubeVirt platform docs related to storage configuration and using the external infrastructure feature to create a hosted cluster.

### CODE REVIEW

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+--memory $MEM \
+--cores $CPU \
+--root-volume-storage-class ocs-storagecluster-ceph-rbd \
+--root-volume-size 64
+----
+
+[#kubevirt-vm-image-caching]
+== Enabling KubeVirt VM image caching
+
+KubeVirt image caching is an advanced feature that you can use to optimize both cluster startup time and storage utilization. This feature requires the use of a storage class that is capable of smart cloning and the `ReadWriteMany` access mode.
```
Comment (davidvossel) [inline clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]: would it be possible to link `smart cloning` to the ocp docs here, https://docs.openshift.com/container-platform/4.13/virt/virtual_machines/virtual_disks/virt-cloning-a-datavolume-using-smart-cloning.html ?

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Image caching works as follows:
+
+. The VM image is imported to a PVC that is associated with the hosted cluster.
+. A unique clone of that PVC is created for every KubeVirt VM that is added as a worker node to the cluster. 
+
+Image caching reduces VM startup time by requiring only a single image import. It can further reduce overall cluster storage usage when the storage class supports copy-on-write cloning.
+
+To enable image caching, during cluster creation, use the `hcp` command line interface and the `--root-volume-cache-stragegy=PVC` argument, as shown in the following example:
```
Comment (davidvossel) [inline clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]: the cli arg here has a typo, it should be `--root-volume-cache-strategy`

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]: Yes - done ✅

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc (hunk 2 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]: Good catch! It should be fixed now.

**Review by davidvossel — APPROVED** [review]
/lgtm

thanks!

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+After you create the hosted cluster, the `guest-sc1` and `guest-sc2` storage classes are visible within the hosted cluster. When you create a PVC within the hosted cluster that uses one of those storage classes, KubeVirt CSI provisions that volume by using the infrastructure storage class mapping that you configured during cluster creation.
+
+*Note:* KubeVirt CSI supports mapping only a infrastructure storage class that is capable of `ReadWriteMany` (RWX) access.
+
+[#kubevirt-vm-root-volume-config]
+== Configuring KubeVirt VM root volume
+
+At cluster creation time, you can configure the storage class that is used to host the KubeVirt VM root volumes by using the `hcp` command line interface and the `--root-volume-storage-class` argument. You can configure the size of the volume by using the `--root-volume-size` argument.
+
+To set a custom storage class and volume size for KubeVirt VMs, see the following example. In the example, the result is a hosted cluster with VMs that are hosted on 64Gi PVCs that are hosted by the `ocs-storagecluster-ceph-rdb` storage class.
```
Comment (oafischer) [inline clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]:
```suggestion
To set a custom storage class and volume size for KubeVirt VMs, see the following example. In the example, the result is a hosted cluster with VMs that are hosted on 64Gi PVCs that are hosted by the `ocs-storagecluster-ceph-rdb` storage class:
```

File: clusters/hosted_control_planes/configuring_storage_kubevirt.adoc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+= Configuring storage for hosted control planes on OpenShift Virtualization
+
+If no advanced configuration is provided, the default storage class is used for the KubeVirt virtual machine (VM) images, the KubeVirt CSI mapping, and the etcd volumes.
+
+[#storageclass-mapping]
+== Mapping KubeVirt CSI storage classes
+
+KubeVirt CSI permits any infrastructure storage class with the `ReadWriteMany` access mode to be exposed to the hosted cluster. This mapping of infrastructure cluster storage class to hosted cluster storage class can be configured during cluster creation by using the `hcp` command line interface and the `--infra-storage-class-mapping` argument.
+
+To map infrastructure storage classes to hosted storage classes, see the following example. The example shows how to map two infrastructure storage classes called `infra-sc1` and `infra-sc2` to hosted storage classes called `guest-sc1` and `guest-sc2`. The `--infra-storage-class-mapping` argument can be used multiple times within the `create` command.
```
Comment (oafischer) [inline clusters/hosted_control_planes/configuring_storage_kubevirt.adoc]:
```suggestion
To map infrastructure storage classes to hosted storage classes, see the following example. The example shows how to map two infrastructure storage classes called `infra-sc1` and `infra-sc2` to hosted storage classes called `guest-sc1` and `guest-sc2`. The `--infra-storage-class-mapping` argument can be used multiple times within the `create` command:
```

**Review by oafischer — APPROVED** [review]
Small suggestions, otherwise looks great!

**Comment by lahinson** [comment]
@oafischer I incorporated your suggestions. Thanks for the review!

**Pull request merged**